src/hamster/lib/datetime.py

```diff
@@ -591,26 +591,42 @@ def from_start_end(cls, start, end=None):
         and to handle either hdays or datetimes.
         """
         if isinstance(start, Range):
-            assert end is None, "range and end are mutually exclusive"
-            range = start
+            assert end is None, "end cannot be passed together with a Range"
+            return cls(start.start, start.end)
+        elif (start is None) or isinstance(start, datetime):
+            # This one must come first,
+            # because inheritance order is datetime < date < pdt.date.
```

Comment:
As pointed out by @mwilck, I think this comment should be:
```suggestion
            # because inheritance order is datetime < pdt.datetime.
```

Code would not need to be changed.